presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -156,6 +163,186 @@ protected QueryRunner createQueryRunner()
         return HiveQueryRunner.createQueryRunner(ORDERS, CUSTOMER);
     }
 
+    @Test
+    public void testLackOfPartitionFilterNotAllowed() throws Exception
+    {
+        Session admin = Session.builder(getQueryRunner().getDefaultSession())
+                .setIdentity(new Identity("hive", Optional.empty(), ImmutableMap.of("hive", new SelectedRole(SelectedRole.Type.ROLE, Optional.of("admin")))))
+                .setCatalogSessionProperty("hive", "query_partition_filter_required", "true")
+                .build();
+
+        assertUpdate(
+                admin,
+                "create table partition_test(\n"
+                + "id integer,\n"
+                + "a varchar,\n"
+                + "b varchar,\n"
+                + "ds varchar)"
+                + "WITH (format='PARQUET', partitioned_by = ARRAY['ds'])");
+        assertUpdate(admin, "insert into partition_test(id,a,ds) values(1, 'a','a')", 1);
+        getQueryRunner().installPlugin(new TestingEventListenerPlugin(generatedEvents));
+        generatedEvents.initialize(2);
+        assertQueryFails(admin, "select id from partition_test where a = 'a'", "Filter required on tpch\\.partition_test for at least one partition column.*");
```

Comment:
The above query fails to generate a QueryCompletedEvent. Or maybe I'm not testing this correctly? We rely on the event to occur so that we can log the exception (and know which queries will fail this partition filter check before we enable this feature). Any ideas, @dain?

Reply:
Moved test and fix to https://github.com/prestosql/presto/pull/2544